sdk/storage/storage-queue/test/utils/recorder.ts

```diff
@@ -0,0 +1,266 @@
+          const [key, value] = pair.split(": ");
+          responseHeaders[key] = value;
+        }
+
+        const parsedUrl = queryString.parseUrl(req.url);
+
+        recordings.push({
+          method: req.method,
+          url: parsedUrl.url,
+          query: parsedUrl.query,
+          requestBody: (data instanceof Blob) ? await blobToString(data) : data,
+          status: req.status,
+          response: (req.response instanceof Blob) ? await blobToString(req.response) : req.response,
+          responseHeaders: responseHeaders
+        });
+      }
+    },
+    playback: function() {
+      xhr = nise.fakeXhr.useFakeXMLHttpRequest();
+      recordings = (window as any).__json__["recordings/" + fp].recordings;
+      uniqueTestInfo = (window as any).__json__["recordings/" + fp].uniqueTestInfo;
+
+      xhr.onCreate = function(req: any) {
+        const reqSend = req.send;
+        req.send = async function(data: any) {
+          reqSend.call(req, data);
+
+          const parsedUrl = queryString.parseUrl(req.url);
+          let formattedRequest = {
+            method: req.method,
+            url: parsedUrl.url,
+            query: parsedUrl.query,
+            requestBody: (data instanceof Blob) ? await blobToString(data) : data
+          };
+
+          let recordingFound = false;
+          for (let i = 0; !recordingFound && i < recordings.length; i++) {
+            if (matchRequest(recordings[i], formattedRequest)) {
+              let status = recordings[i].status;
+              let responseHeaders = recordings[i].responseHeaders;
+              let response = recordings[i].response;
+              setTimeout(() => req.respond(status, responseHeaders, response));
+              recordings.splice(i, 1);
+              recordingFound = true;
+            }
+          }
+
+          if (!recordingFound) {
+            throw new Error("No match for request " + JSON.stringify(formattedRequest, null, " "));
+          }
+        }
+      }
+
+      // We're not matching request headers
+      function matchRequest(recording: any, request: any) {
+        for (let param in recording.query) {
+          if (recording.query[param] !== request.query[param] && param !== "_") {
+            return false;
+          }
+        }
+
+        for (let param in request.query) {
+          if (recording.query[param] === undefined) {
+            return false;
+          }
+        }
+
+        return (
+          recording.method === request.method &&
+          recording.url === request.url &&
+          recording.requestBody === request.requestBody
+        );
+      }
+    },
+    stop: function() {
+      console.log(JSON.stringify({
+        writeFile: true,
+        path: "./recordings/" + fp,
+        content: { recordings: recordings, uniqueTestInfo: uniqueTestInfo }
+      }));
+      xhr.restore();
+    }
+  };
+}
+
+export function record(this: any, folderpath: string, testTitle?: string) {
```

Comment:
Two questions
- Do we need to pass the `folderpath` ? 
[ we can remove this argument and call `this.fullTitle()` inside the function ?]
-  Why do we need this `testTitle` (optional argument) ?
[ I don't see any references using this argument in the tests ]

Reply:
- The `fullTitle` method is available in `this` object within the `describe` block (let's call it `this1`). We're passing the `this` object from `beforeEach` block (call it `this2`) to obtain the test name (and skip it if necessary). The `fullTitle` method is only available in `this1`. I believe there's an array we can use in `this2` to get the full path of the test (in case we had nested `describe` blocks: something like `[describe1, describe2, ..., testName]`). We can use that if you think it's better, but it would be safer to concatenate the elements of the array (`recordings/describe1/describe2/.../recording_testName`) instead of assuming we'll always have only 2 elements.

- In fact, it's not used in storage-queue. We use it in storage-file and storage-blob packages to create recordings for `before` and `after` blocks.

Reply:
Complementing the response for the first question: I believe we would need to deal with `before` and `after` blocks separately if we decided to follow the array approach. Their `this` object may have different properties.

Reply:
I like the concatenation option better.
How about turning the array `[describe1, describe2, ..., testName]` into (`recordings/describe1_describe2_..../recording_testName`)  (just avoiding too many directories)
and removing the argument `folderpath` altogether from the `record` module ?

Reply:
>Complementing the response for the first question: I believe we would need to deal with before and after blocks separately if we decided to follow the array approach. Their this object may have different properties.

I don't mind ignoring if it is complicated.

Reply:
Will need to document this with a proper explanation on how to pass the attributes. [for future purposes]

Reply:
Done. I found out `fullTitle` already returns the full hierarchy of describe blocks. We'll need to update the `after` and `before` recording names in other packages (`after all hook` and `before all hook` are the new names, I think), but that's simple.